cmd/azure-config-credentials-injector/main.go

```diff
@@ -106,6 +121,17 @@ func prepareCloudConfig(cloudConfig map[string]interface{}, clientId string, cli
 		}
 	}
 
+	// Azure workload identity should take precedence, if available.
+	if len(tenantId) != 0 && len(federatedTokenFile) != 0 {
+		cloudConfig[tenantIdConfigKey] = tenantId
+		cloudConfig[aadFederatedTokenFileConfigKey] = federatedTokenFile
+		cloudConfig[useFederatedWorkloadIdentityExtensionConfigKey] = true
+	} else {
+		// Workload identity failed, but we have a secret.
+		klog.V(4).Info("%s env variable is set, client secret authentication will be used", clientSecretEnvKey)
+		cloudConfig[clientSecretCloudConfigKey] = clientSecret
```

Comment:
do we need to check `len(clientSecret) > 0` ?

Reply:
Good point, I'm not sure if we want to pass through `""` - I don't think CCO should let that through. But it does make sense to check for this. But if we do that with secret we should do that with all other values to be consistent. See my latest commit.